src/Polysemy/Internal/Scoped.hs

```diff
@@ -25,26 +25,69 @@ import Polysemy.Internal.Combinators (transform)
 --
 -- The constructors are not intended to be used directly; the smart constructor 'scoped' is used like a local
 -- interpreter for @effect@.
-data Scoped (resource :: Type) (effect :: Effect) :: Effect where
-  Run :: ∀ resource effect m a . resource -> effect m a -> Scoped resource effect m a
-  InScope :: ∀ resource effect m a . (resource -> m a) -> Scoped resource effect m a
+-- 'scoped' takes an argument of type @param@, which will be passed through to the interpreter, to be used by the
+-- resource allocation function.
+data Scoped (param :: Type) (resource :: Type) (effect :: Effect) :: Effect where
+  Run :: ∀ param resource effect m a . resource -> effect m a -> Scoped param resource effect m a
+  InScope :: ∀ param resource effect m a . param -> (resource -> m a) -> Scoped param resource effect m a
+
+-- |A convenience alias for a scope without parameters.
+type Scoped_ resource effect =
+  Scoped () resource effect
 
 -- |Constructor for 'Scoped', taking a nested program and transforming all instances of @effect@ to
--- @Scoped resource effect@.
+-- @'Scoped' param resource effect@ and wrapping the result with 'InScope'.
+--
+-- This allows the effective interpreter to bracket the nested program with a resource from a distance.
+--
+-- The value @param@ is passed to the interpreter.
 scoped ::
+  ∀ param resource effect r .
+  Member (Scoped param resource effect) r =>
+  param ->
+  InterpreterFor effect r
+scoped param main =
+  send $ InScope @param @resource @effect param \ resource ->
+    transform @effect (Run @param resource) main
+{-# inline scoped #-}
+
+-- |Constructor for 'Scoped_', taking a nested program and transforming all instances of @effect@ to
+-- @'Scoped_' resource effect@ and wrapping the result with 'InScope'.
+--
+-- This allows the effective interpreter to bracket the nested program with a resource from a distance.
+scoped_ ::
```

Comment:
Perhaps `scoped_` docs should be something like
```
-- |Constructor for 'Scoped_'. Similar to 'scoped', but <explain differences and importantly give an example when you want this>
```
?
Or perhaps the example would be better attached to `type Scoped_`?

Reply:
I added an extended example to `Scoped` and referenced it here. satisfactory?

Reply:
Awesome